app/selfserve/module/Olcs/src/Controller/OperatorRegistrationController.php

```diff
@@ -50,8 +50,18 @@ public function addAction(): Response|ViewModel
                         'pageTitle' => 'user-registration.page.check-email.title'
                     ]);
                 }
+                $result = $response->getResult();
+
+                if (!empty($result['messages'])) {
+                    $form->setMessages(
+                        [
+                            'main' => $result['messages'],
+                        ]
+                    );
+                } else {
+                    $this->flashMessengerHelper->addErrorMessage('unknown-error');
+                }
```

Comment:
You don't need the else here - can just allow it to go straight through as previously, with the more specific error
```suggestion
                }

                $this->flashMessengerHelper->addErrorMessage('There was an error registering your account. Please try again.');
```